src/hooks/xcm/useXcmBridge.ts

```diff
@@ -127,36 +128,96 @@ export function useXcmBridge(selectedToken: Ref<ChainAsset>) {
   };
 
   const setRelaychainBal = async (): Promise<void> => {
-    if (!relayChainApi || !selectedToken.value) return;
+    if (
+      !relayChainApi ||
+      !selectedToken.value ||
+      selectedToken.value.metadata === null ||
+      isH160.value
+    ) {
+      return;
+    }
     await relayChainApi.isReady();
     const rawBalance = await getRelayChainNativeBal();
     const decimals = Number(String(selectedToken.value.metadata.decimals));
     const balance = ethers.utils.formatUnits(rawBalance, decimals).toString();
     relaychainBal.value = Number(balance);
   };
 
-  const checkIsEnoughEd = (amount: number): boolean => {
+  const checkIsEnoughEd = async (amount: number): Promise<boolean> => {
     const relaychainMinBal = existentialDeposit.value?.relaychainMinBal;
     if (!relaychainMinBal) return false;
 
     if (isDeposit.value) {
       const relayBalAfterTransfer = relaychainBal.value - amount;
       return relayBalAfterTransfer > relaychainMinBal;
     } else {
-      return relaychainBal.value > relaychainMinBal;
+      // Memo: wait for updating relaychainBalance
+      await wait(500);
+      const relaychainBalance = isH160.value ? evmDestAddressBalance.value : relaychainBal.value;
+      return relaychainBalance > relaychainMinBal;
+    }
+  };
+
+  const checkIsEvmDestAddress = (): boolean => {
+    const isEvmWithdraw = isH160.value && !isDeposit.value;
+    if (isEvmWithdraw && !evmDestAddress.value) {
+      return true;
+    }
+
+    return isH160.value
+      ? isValidAddressPolkadotAddress(evmDestAddress.value)
+      : isValidEvmAddress(evmDestAddress.value);
+  };
+
+  const setErrMsg = async () => {
+    errMsg.value = '';
+
+    if (Number(amount.value) > fromAddressBalance.value) {
+      errMsg.value = t('warning.insufficientBalance');
+      return;
+    }
+    if (isH160.value || !isNativeBridge.value) {
+      // Memo: withdrawal from EVM || Deposit from native to EVM
+
+      if (!evmDestAddress.value) {
+        errMsg.value = '';
+        return;
+      }
+      if (Number(amount.value) > fromAddressBalance.value) {
+        errMsg.value = t('warning.insufficientBalance');
+        return;
+      } else if (!checkIsEvmDestAddress()) {
+        errMsg.value = t('warning.inputtedInvalidDestAddress');
+        return;
+      } else if (!(await checkIsEnoughEd(Number(amount.value)))) {
+        errMsg.value = t('warning.insufficientExistentialDeposit', {
+          network: existentialDeposit.value?.chain,
+        });
+        return;
```

Comment:
perhaps there're so many `return` statement used ?

Reply:
@sirius651 
Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/439/commits/79c42bb8f6d2d2f33614ba8120d4d34434e0c646) commit.